core/src/main/scala/tech/beshu/ror/accesscontrol/blocks/rules/auth/BaseGroupsRule.scala

```diff
@@ -42,7 +42,9 @@ abstract class BaseGroupsRule(val settings: Settings)
     with Logging {
 
   protected def calculateAllowedGroupsForUser(userGroups: UniqueNonEmptyList[Group],
-                                              permittedGroupIds: PermittedGroupIds): Option[UniqueNonEmptyList[Group]]
+                                              groupIds: GroupIds): Option[UniqueNonEmptyList[Group]]
```

Comment:
coutoPL comment migrated from old PR:

_**at the moment it's hard to guess what the groupIds is. Previously, here we had user groups and permitted groups. It was rather easy to distinguish between these two. IMO we should find a better name for groupIds param (and the type too).**_

Reply:
I refactored it a bit
- introduced type `ResolvableGroupsLogic`, with implementations corresponding to `GroupsLogic`
- in BaseGroupsRule.Settings, now there is `resolvableGroupsLogic: ResolvableGroupsLogic` instead of `groupIds: ResolvableGroupIds`
- for example, calculateAllowedGroupsForUser and groupsPotentiallyPermittedByRule are now handled in the base class, because in settings we have more info about the logic, instead of raw list of group ids
- that way, much more logic is generic and defined in the base class `BaseGroupsRule`, instead of implementation for specific rules
- and there is less confusion about the names and purposes